File: frontend/src/api/review.ts
import request from '@/utils/request'

export interface ReviewVO {
  id: number
  shopId: number
  userId: number
  content: string
  rating: number
  images: string
  createTime: string
}

export interface ReviewSubmitParams {
  shopId: number
  userId: number
  content: string
  rating: number
  images?: string
}

export function getReviewList(shopId: number) {
  return request.get<ReviewVO[]>('/reviews', { params: { shopId } })
}

export function submitReview(data: ReviewSubmitParams) {
  return request.post<number>('/reviews', data)
}

File: backend/src/main/java/com/dzdp/entity/Review.java
package com.dzdp.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("tb_review")
public class Review {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long shopId;

    private Long userId;

    private String content;

    private Integer rating;

    private String images;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer isDeleted;
}
